### ISSUE TITLE
--persist --nofilesystem=$HOME seems broken with latest git

### ISSUE THREAD

**Issue by damianatorrpm** [flatpak/flatpak#4030]
I'm wondering if I am doing something wrong with --persist's or if that's broken on latest git
MakeMKV has a "bad" folder $HOME/.MakeMKV
 and by default it has access to home
 on normal flatpak run com.makemkv.MakeMKV the folder($HOME/.MakeMKV) is created (with contents)

1) flatpak --user --persist=.MakeMKV --nofilesystem=$HOME run com.makemkv.MakeMKV causes the $HOME/.MakeMKV to not be created but $HOME/.var/app/com.makemkv.MakeMKV/.MakeMKV instead but no contents and changing makemkv settings also doesn't write to it's config file which is supposed to be in that directory
2) flatpak --user --persist=.MakeMKV --nofilesystem=home run com.makemkv.MakeMKV has no effect at all

**Comment by alexlarsson** [comment]
git master normalizes --filesystem=/path/to/home to --filesystem=home, as the later is a bit more extensible and correct.

**Comment by alexlarsson** [comment]
Ah, i tried the app. The issue is that the app has filesystem=host access, which is greater than filesystem=home. So, nofilesystem=home doesn't do anything. However, nofilesystem=host should work.

**Comment by damianatorrpm** [comment]
Ah, that makes sense, so not a bug than. 
Afaik bwrap can't revoke home if greater is granted.

I noticed that when trying to implement something for: https://github.com/flatpak/flatpak/issues/2485
I added an else clause if (home_access)
https://github.com/flatpak/flatpak/blob/a17224cd08abf1c0f2223d2642a33018f7782fe6/common/flatpak-context.c#L2340-L2355
where I would 
```
// or add tmpfs which is the same
flatpak_exports_add_path_expose_or_hide (exports, FLATPAK_FILESYSTEM_MODE_NONE, dest);        
flatpak_bwrap_add_bind_arg (bwrap, "--bind", src, dest);
```
I mentioned that on irc, but I guess this definetely answers that question as well and that this won't work. 
(since the flatpak_bwrap_add_bind_arg will probably revert the hiding of dest)
